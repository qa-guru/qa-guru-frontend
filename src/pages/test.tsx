import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
