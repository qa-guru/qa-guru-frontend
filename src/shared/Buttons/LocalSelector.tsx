import React from "react";
import { useTranslation } from "react-i18next";
import { Box, IconButton, SvgIcon } from "@mui/material";
import { ReactComponent as EnglishIcon } from "../../assets/icons/english.svg";
import { ReactComponent as RussiaIcon } from "../../assets/icons/russia.svg";

const style = {
  box: { display: "flex", justifyContent: "center" },
};

const LocalSelector = () => {
  const { i18n } = useTranslation();

  const changeLanguage = (lng: string) => {
    i18n.changeLanguage(lng);
  };

  return (
    <Box sx={style.box}>
      <IconButton onClick={() => changeLanguage("en")}>
        <SvgIcon>
          <EnglishIcon />
        </SvgIcon>
      </IconButton>
      <IconButton onClick={() => changeLanguage("ru")}>
        <SvgIcon>
          <RussiaIcon />
        </SvgIcon>
      </IconButton>
    </Box>
  );
};

export default LocalSelector;
