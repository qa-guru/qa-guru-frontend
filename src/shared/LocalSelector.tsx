import React from "react";
import { useTranslation } from "react-i18next";
import { Box, IconButton, SvgIcon } from "@mui/material";
import { ReactComponent as EnglishIcon } from "../icons/english.svg";
import { ReactComponent as RussiaIcon } from "../icons/russia.svg";

const LocalSelector = () => {
  const { i18n } = useTranslation();

  const changeLanguage = (lng: string) => {
    i18n.changeLanguage(lng);
  };

  return (
    <Box sx={{ display: "flex", justifyContent: "center" }}>
      <IconButton onClick={() => changeLanguage("en")}>
        <SvgIcon>
          <EnglishIcon />
        </SvgIcon>
      </IconButton>
      <IconButton onClick={() => changeLanguage("ru")}>
        <SvgIcon>
          <RussiaIcon />
        </SvgIcon>
      </IconButton>
    </Box>
  );
};

export default LocalSelector;
