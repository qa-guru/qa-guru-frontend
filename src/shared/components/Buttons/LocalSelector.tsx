import React from "react";
import { useTranslation } from "react-i18next";
import { Box, IconButton, SvgIcon } from "@mui/material";
import { style } from "./styles";
import { ReactComponent as EnglishIcon } from "../../../assets/icons/english.svg";
import { ReactComponent as RussiaIcon } from "../../../assets/icons/russia.svg";

const LocalSelector = () => {
  const { i18n } = useTranslation();

  const changeLanguage = (lng: string) => {
    i18n.changeLanguage(lng);
  };

  return (
    <Box sx={style.box}>
      <IconButton onClick={() => changeLanguage("en")}>
        <SvgIcon>
          <EnglishIcon />
        </SvgIcon>
      </IconButton>
      <IconButton onClick={() => changeLanguage("ru")}>
        <SvgIcon>
          <RussiaIcon />
        </SvgIcon>
      </IconButton>
    </Box>
  );
};

export default LocalSelector;
