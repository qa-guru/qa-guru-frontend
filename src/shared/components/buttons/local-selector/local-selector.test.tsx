import { MemoryRouter } from "react-router-dom";
import { render } from "test/utils-test";
import LocalSelector from "./local-selector";

describe("LocalSelector", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <LocalSelector />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
