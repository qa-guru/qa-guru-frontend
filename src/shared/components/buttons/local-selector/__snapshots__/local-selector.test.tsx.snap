// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LocalSelector > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1rqcvvl"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          id="Layer_1"
          style="enable-background: new 0 0 512 512;"
          viewBox="0 0 512 512"
          x="0px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
          y="0px"
        >
          <circle
            cx="256"
            cy="256"
            r="256"
            style="fill: #F0F0F0;"
          />
          <g>
            <path
              d="M52.92,100.142c-20.109,26.163-35.272,56.318-44.101,89.077h133.178L52.92,100.142z"
              style="fill: #0052B4;"
            />
            <path
              d="M503.181,189.219c-8.829-32.758-23.993-62.913-44.101-89.076l-89.075,89.076H503.181z"
              style="fill: #0052B4;"
            />
            <path
              d="M8.819,322.784c8.83,32.758,23.993,62.913,44.101,89.075l89.074-89.075L8.819,322.784L8.819,322.784 z"
              style="fill: #0052B4;"
            />
            <path
              d="M411.858,52.921c-26.163-20.109-56.317-35.272-89.076-44.102v133.177L411.858,52.921z"
              style="fill: #0052B4;"
            />
            <path
              d="M100.142,459.079c26.163,20.109,56.318,35.272,89.076,44.102V370.005L100.142,459.079z"
              style="fill: #0052B4;"
            />
            <path
              d="M189.217,8.819c-32.758,8.83-62.913,23.993-89.075,44.101l89.075,89.075V8.819z"
              style="fill: #0052B4;"
            />
            <path
              d="M322.783,503.181c32.758-8.83,62.913-23.993,89.075-44.101l-89.075-89.075V503.181z"
              style="fill: #0052B4;"
            />
            <path
              d="M370.005,322.784l89.075,89.076c20.108-26.162,35.272-56.318,44.101-89.076H370.005z"
              style="fill: #0052B4;"
            />
          </g>
          <g>
            <path
              d="M509.833,222.609h-220.44h-0.001V2.167C278.461,0.744,267.317,0,256,0 c-11.319,0-22.461,0.744-33.391,2.167v220.44v0.001H2.167C0.744,233.539,0,244.683,0,256c0,11.319,0.744,22.461,2.167,33.391 h220.44h0.001v220.442C233.539,511.256,244.681,512,256,512c11.317,0,22.461-0.743,33.391-2.167v-220.44v-0.001h220.442 C511.256,278.461,512,267.319,512,256C512,244.683,511.256,233.539,509.833,222.609z"
              style="fill: #D80027;"
            />
            <path
              d="M322.783,322.784L322.783,322.784L437.019,437.02c5.254-5.252,10.266-10.743,15.048-16.435 l-97.802-97.802h-31.482V322.784z"
              style="fill: #D80027;"
            />
            <path
              d="M189.217,322.784h-0.002L74.98,437.019c5.252,5.254,10.743,10.266,16.435,15.048l97.802-97.804 V322.784z"
              style="fill: #D80027;"
            />
            <path
              d="M189.217,189.219v-0.002L74.981,74.98c-5.254,5.252-10.266,10.743-15.048,16.435l97.803,97.803 H189.217z"
              style="fill: #D80027;"
            />
            <path
              d="M322.783,189.219L322.783,189.219L437.02,74.981c-5.252-5.254-10.743-10.266-16.435-15.047 l-97.802,97.803V189.219z"
              style="fill: #D80027;"
            />
          </g>
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
        </svg>
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          clip-rule="evenodd"
          fill-rule="evenodd"
          image-rendering="optimizeQuality"
          shape-rendering="geometricPrecision"
          text-rendering="geometricPrecision"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill-rule="nonzero"
          >
            <path
              d="M256 0c70.684 0 134.689 28.664 181.012 74.987C483.336 121.311 512 185.316 512 256c0 70.684-28.664 134.689-74.988 181.013C390.689 483.336 326.684 512 256 512c-70.677 0-134.689-28.664-181.013-74.987C28.664 390.689 0 326.676 0 256c0-70.684 28.664-134.689 74.987-181.013C121.311 28.664 185.316 0 256 0z"
              fill="#4D4D4D"
            />
            <path
              d="M256.001 19.597c65.278 0 124.382 26.466 167.162 69.242 42.776 42.78 69.242 101.884 69.242 167.162S465.939 380.384 423.16 423.16c-42.777 42.78-101.881 69.246-167.159 69.246-65.278 0-124.382-26.466-167.162-69.243-42.777-42.779-69.243-101.884-69.243-167.162S46.062 131.619 88.839 88.839c42.78-42.776 101.884-69.242 167.162-69.242z"
              fill="#fff"
            />
            <path
              d="M472.405 256.232H39.597l-.003-.231c0-119.52 96.886-216.407 216.407-216.407 119.517 0 216.407 96.887 216.407 216.407l-.003.231z"
              fill="#fff"
            />
            <path
              d="M39.597 256.232h432.808c-.124 119.411-96.963 216.176-216.404 216.176-119.444 0-216.28-96.765-216.404-216.176z"
              fill="#D52B1E"
            />
            <path
              d="M52.038 183.55h407.926c8.048 22.652 12.444 47.036 12.444 72.451 0 25.586-4.452 50.131-12.606 72.915H52.2c-8.154-22.784-12.606-47.329-12.606-72.915 0-25.415 4.396-49.799 12.444-72.451z"
              fill="#0039A6"
            />
          </g>
        </svg>
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
